lib/components/mobile/edit-search-button.js

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+
+import * as formActions from '../../actions/form'
+import * as uiActions from '../../actions/ui'
+
+/**
+ * Renders the "Edit" or "Back to search" button in mobile result views
+ * that takes the user back to the mobile search screen.
+ */
+class EditSearchButton extends Component {
+  static propTypes = {
+    clearActiveSearch: PropTypes.func,
+    setItineraryView: PropTypes.func,
+    setMobileScreen: PropTypes.func
+  }
+
+  _handleClick = () => {
+    const { clearActiveSearch, setItineraryView, setMobileScreen } = this.props
+
+    // Reset itinerary view state to show the list of results *before* clearing the search.
+    // (Otherwise, if the map is expanded, the search is not cleared.)
+    setItineraryView(uiActions.ItineraryView.LIST)
+    clearActiveSearch()
+    setMobileScreen(uiActions.MobileScreens.SEARCH_FORM)
+  }
```

Comment:
It would simplify this component a fair amount if this method were converted into a connected action. This comment in non-blocking.

Reply:
Thanks for the suggestion, updated in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/43fef77367b6d788c401198500108f49fff1f355.